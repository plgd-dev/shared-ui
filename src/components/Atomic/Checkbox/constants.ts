export const checkboxValues = {
    ON: 'on',
    OFF: 'off',
} as const
