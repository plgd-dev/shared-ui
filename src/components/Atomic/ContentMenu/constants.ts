export const statuses = {
    SUCCESS: 'success',
    ERROR: 'error',
    WARNING: 'warning',
} as const
