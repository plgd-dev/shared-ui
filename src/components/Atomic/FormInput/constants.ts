export const inputSizes = {
    BIG: 'big',
    NORMAL: 'normal',
    SMALL: 'small',
} as const

export const inputAligns = {
    LEFT: 'left',
    RIGHT: 'right',
} as const
