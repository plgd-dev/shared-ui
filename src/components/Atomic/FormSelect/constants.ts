export const selectSizes = {
    NORMAL: 'normal',
    SMALL: 'small',
} as const

export const selectAligns = {
    LEFT: 'left',
    RIGHT: 'right',
} as const
