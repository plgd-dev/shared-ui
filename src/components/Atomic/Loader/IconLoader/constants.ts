export const types = {
    PRIMARY: 'primary',
    SECONDARY: 'secondary',
    TERTIARY: 'tertiary',
    Filter: 'filter',
} as const
