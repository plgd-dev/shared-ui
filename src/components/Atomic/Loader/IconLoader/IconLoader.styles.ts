import { keyframes } from '@emotion/css'
import { css } from '@emotion/react'
import { IconLoaderType } from './IconLoader.types'
import { types } from './constants'
import { colors } from '../../_utils/colors'

const spin = keyframes`
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(180deg);
  }
  100% {
    transform: rotate(360deg);
  }
`

const getColorByType = (type: IconLoaderType) => {
    switch (type) {
        case types.PRIMARY:
            return css`
                color: ${colors.neutral000};
            `
        case types.SECONDARY:
            return css`
                color: ${colors.primary};
            `
        case types.TERTIARY:
            return css`
                color: ${colors.neutral900};
            `
    }
}

export const loadingIcon = (size: number, type: IconLoaderType) => css`
    width: ${size}px;
    height: ${size}px;
    overflow: hidden;
    display: block;
    animation: 0.75s linear 0s infinite normal both running ${spin};
    ${getColorByType(type)};
`
