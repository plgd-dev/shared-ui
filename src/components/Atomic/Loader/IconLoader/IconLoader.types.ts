import { Props as IconProps } from '../../Icon/Icon.types'
import { types } from './constants'

export type IconLoaderType = (typeof types)[keyof typeof types]

export type Props = {
    center?: boolean
    type?: IconLoaderType
} & Omit<IconProps, 'icon'>

export const defaultProps = {
    type: types.PRIMARY,
}
