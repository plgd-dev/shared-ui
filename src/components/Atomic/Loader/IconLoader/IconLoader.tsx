import { FC } from 'react'
import Icon from '../../Icon'
import { Props, defaultProps } from './IconLoader.types'
import * as styles from './IconLoader.styles'

const defaultSize = 20

const IconLoader: FC<Props> = (props) => {
    const { type, className, ...rest } = { ...defaultProps, ...props }
    return (
        <span className={className} css={styles.loadingIcon(props.size || defaultSize, type)}>
            <Icon {...rest} icon='loader' ref={undefined} size={props.size || defaultSize} />
        </span>
    )
}

IconLoader.displayName = 'IconLoader'
IconLoader.defaultProps = defaultProps

export default IconLoader
