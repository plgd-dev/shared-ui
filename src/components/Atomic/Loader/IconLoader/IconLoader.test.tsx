import { render } from '../../../../../test/jest-wrapper'
import IconLoader from './IconLoader'
import { types } from './constants'

describe('<IconLoader>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <IconLoader />
                <IconLoader type={types.PRIMARY} />
                <IconLoader type={types.SECONDARY} />
                <IconLoader type={types.TERTIARY} />
                <IconLoader type={types.Filter} />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
