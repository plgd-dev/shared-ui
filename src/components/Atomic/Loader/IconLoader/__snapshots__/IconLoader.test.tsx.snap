// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLoader> render correctly - snapshot 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(180deg);
    -moz-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  width: 20px;
  height: 20px;
  overflow: hidden;
  display: block;
  -webkit-animation: 0.75s linear 0s infinite normal both running animation-0;
  animation: 0.75s linear 0s infinite normal both running animation-0;
  color: #FFFFFF;
}

.emotion-2 {
  width: 20px;
  height: 20px;
  overflow: hidden;
  display: block;
  -webkit-animation: 0.75s linear 0s infinite normal both running animation-0;
  animation: 0.75s linear 0s infinite normal both running animation-0;
  color: #2261AE;
}

.emotion-3 {
  width: 20px;
  height: 20px;
  overflow: hidden;
  display: block;
  -webkit-animation: 0.75s linear 0s infinite normal both running animation-0;
  animation: 0.75s linear 0s infinite normal both running animation-0;
  color: #191A1A;
}

.emotion-4 {
  width: 20px;
  height: 20px;
  overflow: hidden;
  display: block;
  -webkit-animation: 0.75s linear 0s infinite normal both running animation-0;
  animation: 0.75s linear 0s infinite normal both running animation-0;
}

<div>
    <span
      class="emotion-0"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 19"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
          fill="currentcolor"
        />
      </svg>
    </span>
    <span
      class="emotion-0"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 19"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
          fill="currentcolor"
        />
      </svg>
    </span>
    <span
      class="emotion-2"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 19"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
          fill="currentcolor"
        />
      </svg>
    </span>
    <span
      class="emotion-3"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 19"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
          fill="currentcolor"
        />
      </svg>
    </span>
    <span
      class="emotion-4"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 19"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
          fill="currentcolor"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
